tests/shared/test_port_handler.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+
+from ert_shared import port_handler, Service
+
+
+def test_assign_port():
+    evaluator_port = port_handler.assign_port(service=Service.evaluator)
+    storage_port = port_handler.assign_port(service=Service.storage)
+
+    assert storage_port.port != evaluator_port.port
+    assert storage_port.port in port_handler._range
+    assert storage_port.host == evaluator_port.host
+    assert storage_port.host == port_handler._get_ip_address()
+
+
+def test_assign_custom_port():
+    custom_port = 50000
+    evaluator_port = port_handler.assign_port(
+        service=Service.evaluator, port=custom_port
+    )
+    assert evaluator_port.port == custom_port
+
+
+def test_assign_port_invalid_host():
+    custom_host = "dummy_host"
+
+    with pytest.raises(Exception) as exc_info:
```

Comment:
In stead of keeping the exception and asserting on the string, pytest offers a `match` parameter that regex compares the excepted output with the exception ouput.
```suggestion
    with pytest.raises(Exception, match=f"Name or service not known ({custom_host})"):
```

Reply:
Nice! :smile:  But i think i just need to remove the assertions as linux / osx both gives different error codes along with different error msgs so it is hard to assert apparently ... :disappointed: